### PR TITLE
Fix up received stanza size metric

### PR DESCRIPTION
**Pull request by igors**
This PR addresses #1614

Proposed changes include:

Use cumulative chunk size when reporting the size of received stanzas. Chunk size "accumulates" until exml parses out a full stanza.

This is still imperfect since a single chunk can potentially contain more than one stanza but it's a much more useful approximation than never reporting anything above MTU size.

### CODE REVIEW

File: src/ejabberd_receiver.erl [hunk 1]
```diff
@@ -123,6 +124,7 @@ init([Socket, SockMod, Shaper, MaxStanzaSize]) ->
                 sock_mod = SockMod,
                 shaper_state = ShaperState,
                 max_stanza_size = MaxStanzaSize,
+                stanza_chunk_size = 0,
```
Comment (arcusfelis) [inline src/ejabberd_receiver.erl]: it should be set to 0, each time we reset stream.

Comment (igors) [reply src/ejabberd_receiver.erl]: Thanks! Let me know if c107deb is not what you had in mind.

**Comment by igors** [comment]
Would you rather I return a new State from reset_parser? I've no problem changing that if it's preferred from your POV - just let me know if you'd like a new commit or a rebase.

**Comment by fenek** [comment]
Actually it is a pretty good idea to have `reset_parser` accept only `State` as parameter and return new state. Please change it as a new commit to this branch. :)

**Comment by igors** [comment]
Take a look at c19f52d ;)

**Comment by fenek** [comment]
Thanks. :) Could you please rename `ParserState` to something like `StateAfterReset`? Currently the name suggests that this could be some new type of state.

**Comment by igors** [comment]
Done

**Comment by fenek** [comment]
Looks good now! Thank you. :)

**Pull request merged**